tests/clientEncryption/clientEncryption-decrypt-001.phpt

```diff
@@ -0,0 +1,27 @@
+--TEST--
+MongoDB\Driver\clientEncryption::decrypt()
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_libmongocrypt(); ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_server_version('<', '3.6'); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$key = base64_decode('Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk');
+
+$manager = new MongoDB\Driver\Manager(URI);
+$clientEncryption = $manager->clientEncryption(['keyVaultNamespace' => 'default.keys', 'kmsProviders' => ['local' => ['key' => new MongoDB\BSON\Binary($key, 0)]]]);
+
+$key = $clientEncryption->createDataKey('local');
+
+$encrypted = $clientEncryption->encrypt('top-secret', ['keyId' => $key, 'algorithm' => 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic']);
```

Comment:
From the spec, there appears to be [two algorithms](https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#algorithm). Should we consider defining these algorithms as constants on the ClientEncryption class? 

If so, are these only used for ClientEncryption? The EncryptOpts in the spec appear to only be for the explicit API. I imagine auto encryption can only use deterministic encryption, so perhaps it's not even an option there since there are only two supported algorithms at present (deterministic and random).

Reply:
The "encrypt" option for `jsonSchema` also allows for an `algorithm` option, which accepts both `random` and `deterministic` algorithms. That said, putting these constants in `ClientEncryption` is probably the best place to store them, as they may be confusing if added to the `Manager` class itself.

I was considering adding them as you really don't want to type these out as strings every time you need to use them, but pretty much forgot. I'll add them, but will replace the `-Deterministic` portion with `_DETERMINISTIC` and `_RANDOM` to adhere to the constant naming pattern that we use. That would make these constants `AEAD_AES_256_CBC_HMAC_SHA_512_DETERMINISTIC` and `AEAD_AES_256_CBC_HMAC_SHA_512_RANDOM`. Still a handful, but at least code completion can help the user pick an algorithm without adding typos. 👍